rcl/src/rcl/arguments.c

```diff
@@ -93,10 +93,10 @@ rcl_arguments_get_param_files(
     if (NULL == (*parameter_files)[i]) {
       // deallocate allocated memory
       for (int r = i; r >= 0; --r) {
-        if (NULL == (*parameter_files[r])) {
-          break;
+        if (NULL == (*parameter_files)[r]) {
+          continue;
         }
```

Comment:
You can take advantage that deallocating `NULL` is a noop, that makes things easier.

```suggestion
```

Reply:
Now that I think about it, the iteration stops on the first NULL, so actually this check is not needed because previous allocated elements are guaranteed to be not NULL.